cmd/zfs/zfs_main.c

```diff
@@ -6195,14 +6195,19 @@ unshare_unmount(int op, int argc, char **argv)
 	char sharesmb[ZFS_MAXPROPLEN];
 
 	/* check options */
-	while ((c = getopt(argc, argv, op == OP_SHARE ? "a" : "af")) != -1) {
+	while ((c = getopt(argc, argv, op == OP_SHARE ? ":a" : "af")) != -1) {
```

Comment:
So my reading of `zfs(8)` is that this new behavior isn't described in either the `share` or `unshare` sections of the man page.  Although it clearly exists for `zfs share -a`, I have no idea how I'd know that `zfs share <nfs|smb> -a` is supported.  We should make this clearer.